### ISSUE TITLE
Document event defect

### ISSUE THREAD

**Issue by berryd** [jdewit/bootstrap-timepicker#286]
In showWidget we have this:

```
$(document).on('mousedown.timepicker', function (e) {
        // Clicked outside the timepicker, hide it
        if ($(e.target).closest('.bootstrap-timepicker-widget').length === 0) {
          self.hideWidget();
        }
      });
```

If the selector were '.bootstrap-timepicker', it would behave similarly to the datepicker in that the input field can receive focus without closing the component.  

Additionally, if multiple timepickers are opened, any click inside of any timepicker will not close the other and this will be processed for each event appended.  For the event to correlate to the instance of the timepicker, it should look within the scope of the current instance, such as this.$element.parent()...
this.$element.parent().find(e.target).length===0

If making the event a singleton, broadcast an event on the e.target which closes it rather than having a reference to 'self', and of course, ensure only one event listener can be set.

Basically I'm suggesting the following:  
set an event on the element which listens to something like 'close.timepicker'

```
$(document).on('mousedown.timepicker', function (e) {
        var $el = this.$element.parent();
        if ($el.find(e.target).length === 0) {
          $el.trigger('close.timepicker');
        }
      });
```

**Comment by mrhota** [comment]
I'm not certain I know what the defect is. What action have you performed? What was the expected behavior? What was the observed behavior?
